### ISSUE TITLE
Seek bar on iPhone  view

### ISSUE THREAD

**Issue by KimPatch** [DukeLivingHistory/RLH-Site-Code#44]
There's a bug on the phone view staging site:
When I go to the end of the file, the seek bar only goes about 1/4 way across rather than all the way at the end of the field. It works, but is shortened.
(Note that this is the case only in phone view, tablet view is fine)

**Comment by KimPatch** [comment]
(iPhone 6/iOS 11.2.2) I was seeing the phone bug consistently yesterday, but it's working fine today – maybe it was a caching issue. The seek bar is still not working properly, though. In landscape mode I touch the seek bar and it's pretty immediately responsive. In portrait mode it often doesn't respond, and sometimes it looks like the long press is selecting. It's not always responding in portrait mode, but in landscape mode it seems much worse.
![image](https://user-images.githubusercontent.com/7329361/36575129-369cbbe4-1817-11e8-9296-ff613cb75071.png)

**Comment by ethanbutler** [comment]
Glad to hear the width is looking better. Regarding the dragging behavior, this might be an issue with Able Player itself. Can you check out [this demo](http://ableplayer.github.io/ableplayer/demos/video1.html) and see if it works there? It doesn't seem to respond for me.

This might be because the library wasn't intended to be used for smaller touch devices. I can look into this, but I'd recommend waiting until the next version of Able Player is out.

**Comment by KimPatch** [comment]
The Able Player demo works well – seek bar is very responsive to touch (just to be clear, not drag, just touch to change the location of the button) even though the video shows up very small on the screen of the iPhone.

**Comment by ethanbutler** [comment]
Great, thanks for the clarification – I had only been trying drags. It looks like this stems from some CSS I added in order for the seekbar to go full-width (disabling it via Chrome's Inspector fixed the problem). I'll look through the AP source code to see if I can figure out why this is having this effect.

**Comment by ethanbutler** [comment]
@KimPatch – update on this. After spending some time going through the Able Player source code, there isn't an easy fix for this. There's a lot of math in the Able Player source code that revolves around the seekbar being a specific width. As far as I can tell, the issue here is that whenever the video is playing, the player is constantly updating the UI. When the width isn't what's expected, the player resets itself to the previous value, which is why the seekbar doesn't respond. It would take a pretty extensive rewrite in order to fix this behavior.

I've restored the functionality by rolling back to a commit without the full-width.

I know that there's some work being done on that project that might make this possible in the future. I'm marking this as wontfix for the time being.

**Comment by KimPatch** [comment]
Testing this morning it looks like it's not rolled back – and I'm not sure it should be. 

**Question:** Can you roll this back just for _iPhone_ _portrait_ mode? The larger seek bar is better and works in iPhone landscape mode. In any case, I think it definitely _shouldn't_ be rolled back if we can't separate out the iPhone. This is a big improvement for seeing and works well on the computer and Tablet. I'd rather give up the secondary way to navigate in portrait mode on the iPhone. Users can still navigate  using the text.

**Comment by ethanbutler** [comment]
I just pushed a fix that hides the seekbar from the medium/small views on portrait, but keeps it for the large view on that (and any larger screen sizes).

**Comment by KimPatch** [comment]
Sounds a good solution, but there is something amiss – the phone small view landscape seek bar is foreshortened and the phone small view portrait is now on the left side instead of the right. The phone Medium view Portrait still has the seek bar, and in going back and forth between the small view without the seek bar and the medium view I think we should just keep it like it was. The visual information is very useful – you can use the text to navigate up to the next hit. I think users will assume it's a static seek bar. 

Meanwhile, we can keep tabs on a better future solution from Able Player.

**Comment by ethanbutler** [comment]
10-4. I've rolled it back so that the seekbar is hidden for sm/md portrait phone views, and visible on every other view.

**Comment by KimPatch** [comment]
Shoot – I think we crossed wires on communication there – I was suggesting putting it back the way it was  with the seek bar on all views, thinking that users are going to just think it's a static seekbar on the iPhone portrait views but even so the visual information is useful enough that we should leave it. 

Also, some other stuff got rolled back in the latest fix that shouldn't have. We eliminated the control buttons except for the seek bar on the iPhone view and did stop/start when you touch anywhere on the video. Currently in the horizontal view I can't stop/start by touching the picture and the buttons are there and shouldn't be. We'd had it so you were able to touch the picture to stop/start in every phone view.

So the bottom line is I think we should put this functionality back the way it was before the fix that hid the seek bar, (with the seek bar on every iPhone view and stop/start to touch the picture on every iPhone view).
